packages/shared/src/parsing/code.ts

```
@@ -0,0 +1,58 @@
+ * Handles formats like:
+ * ```language arg1 arg2
+ * code content
+ * ```
+ */
+export function parseCodeBlocks(markdown: string): CodeBlock[] {
+  // Match code blocks including the opening line
+  const codeBlockRegex = /```(.*?)\n([\s\S]*?)```/g
+  const blocks: CodeBlock[] = []
+
+  let match
+  while ((match = codeBlockRegex.exec(markdown)) !== null) {
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)